src/arche/readers/schema.py

```diff
@@ -70,38 +40,58 @@ class Tag(Enum):
     product_price_was_field = (5,)
 
 
-class Tags:
-    values = set([name for name, _ in Tag.__members__.items()])
+class Schema:
+    allowed_tags = set([name for name, _ in Tag.__members__.items()])
+
+    def __init__(self, source: SchemaSource):
+        self.raw: RawSchema = self.read(source)
+        if not self.raw.get("properties", None):
+            raise ValueError("The schema does not have any 'properties'")
+        self.enums: List[str] = self.get_enums()
+        self.tags = self.get_tags(self.raw)
 
-    def __init__(self):
-        self.tagged_fields = defaultdict(list)
+    def json(self):
```

Comment:
maybe `pretty_print` is a better name.
I would expect this method to return a python dict.

Reply:
In this context I think it looks alright -
schema.json() prints a json and the only use I see is to copy paste the output in a `.json` file
and schema.__repr__ (or schema) prints a dict - to actually see it pretty printed